build/miengine.targets

```diff
@@ -7,7 +7,47 @@
 
   <PropertyGroup>
     <CoreCompileDependsOn>GenerateAssemblyInfoFile;$(CoreCompileDependsOn)</CoreCompileDependsOn>
+    <BuildDependsOn Condition="'$(IsCoreClr)' == 'false'">DetermineCompilerBinary;$(BuildDependsOn)</BuildDependsOn>
   </PropertyGroup>
+
+  <UsingTask TaskName="GetRuntime" 
+             TaskFactory="CodeTaskFactory" 
+             AssemblyFile="$(MSBuildToolsPath)\Microsoft.Build.Tasks.v4.0.dll" >
+    <ParameterGroup>
+      <IsMonoRuntime ParameterType="System.Boolean" Output="true" />
+    </ParameterGroup>
+    <Task>
+      <Code Type="Fragment" Language="cs">
+        <![CDATA[
+        IsMonoRuntime = System.Type.GetType("Mono.Runtime") != null;
+        ]]>
+      </Code>
+    </Task>
+  </UsingTask>
+
+  <Target Name="DetermineCompilerBinary">
+    <GetRuntime>
+      <Output PropertyName="IsMonoRuntime" TaskParameter="IsMonoRuntime" />
+    </GetRuntime>
+
+    <Exec Condition="'$(IsMonoRuntime)' == 'true'"
+          ConsoleToMsBuild="true"
+          Command="dirname `which mcs`">
+      <Output TaskParameter="ConsoleOutput" PropertyName="McsPath" />
+    </Exec>
+
+    <PropertyGroup Condition="'$(IsMonoRuntime)' == 'true'">
```

Comment:
Does this just affect what we are compiling with? Or does it affect what we are running with? 

If you are aren't going to use the binaries produced by compiling against the desktop config with 'IsMonoRuntime'='false' to target mono I might suggest instead just having a 'Mono' config. Since you can use the normal config to run on desktop CLR. 
If this is just because you want to be able to build on Linux/OSX for quicker cycles - makes sense.